cli/execute.py

```diff
@@ -59,96 +39,57 @@ def ansible_playbook(playbook, args, inventory="local_hosts"):
     )
-
-    failed_hosts = []
-    unreachable_hosts = []
-
-    if args['verbose']:
-        ansible_cmd = ["ansible-playbook"]
-        if module_path:
-            ansible_cmd.append("-M " + module_path)
-        ansible_cmd.append("-" + "v" * args['verbose'])
-        ansible_cmd.append("-i " + inventory)
-        extra_vars = args['output'] or "<path to settings file>"
-        ansible_cmd.append("--extra-vars @" + extra_vars)
-        ansible_cmd.append(path_to_playbook)
-        print "ANSIBLE COMMAND: " + " ".join(ansible_cmd)
-
-    pb.run()
-
-    hosts = sorted(pb.stats.processed.keys())
-    callbacks.display(callbacks.banner("PLAY RECAP"))
-    playbook_cb.on_stats(pb.stats)
-
-    for host in hosts:
-        status = pb.stats.summarize(host)
-        if status['failures'] > 0:
-            failed_hosts.append(host)
-        if status['unreachable'] > 0:
-            unreachable_hosts.append(host)
-
-    retries = failed_hosts + unreachable_hosts
-
-    if len(retries) > 0:
-        filename = pb.generate_retry_inventory(retries)
-        if filename:
-            callbacks.display(
-                "           to retry, use: --limit @%s\n" % filename)
-
-    for host in hosts:
-        status = pb.stats.summarize(host)
-
-        callbacks.display("%s : %s %s %s %s" % (
-            hostcolor(host, status),
-            colorize('ok', status['ok'], 'green'),
-            colorize('changed', status['changed'], 'yellow'),
-            colorize('unreachable', status['unreachable'], 'red'),
-            colorize('failed', status['failures'], 'red')), screen_only=True)
-
-        callbacks.display("%s : %s %s %s %s" % (
-            hostcolor(host, status, False),
-            colorize('ok', status['ok'], None),
-            colorize('changed', status['changed'], None),
-            colorize('unreachable', status['unreachable'], None),
-            colorize('failed', status['failures'], None)), log_only=True)
-
-    print ""
-    if len(failed_hosts) > 0:
-        raise exceptions.IRPlaybookFailedException(
-            playbook, "Failed hosts: %s" % failed_hosts)
-    if len(unreachable_hosts) > 0:
-        raise exceptions.IRPlaybookFailedException(
-            playbook, "Unreachable hosts: %s" % unreachable_hosts)
+    options = namedtuple('Options', hacked_options.keys())(**hacked_options)
+
+    passwords = dict(vault_pass='secret')
+    variable_manager = VariableManager()
+    variable_manager.extra_vars = args["settings"]
+    inventory = Inventory(loader=loader, variable_manager=variable_manager,
+                          host_list=inventory)
+    variable_manager.set_inventory(inventory)
+
+    pbex = PlaybookExecutor(playbooks=[path_to_playbook], inventory=inventory,
+                            variable_manager=variable_manager, loader=loader,
+                            options=options, passwords=passwords)
+    results = pbex.run()
+    if results:
+        raise exceptions.IRPlaybookFailedException(playbook)
```

Comment:
Results is a return code (0/2/3).
This just means moving the "if results - raise..." to a different module